File: 백준/Bronze/27433. 팩토리얼 2/팩토리얼 2.cc
#include <iostream>

using namespace std;

long long factorial(int N) {
	if (N == 0 || N == 1)
		return 1;
	else
		return N * factorial(N - 1);
}


int main() {
	int N;
	cin >> N;

	cout << factorial(N) << endl;
	return 0;
}

File: 백준/Bronze/27433. 팩토리얼 2/팩토리얼 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int num = Integer.parseInt(br.readLine());

        long result = factorial(num);

        System.out.println(result);
    }

    private static long factorial(int target) {
        if (target == 0 || target == 1)
            return 1;
        return target * factorial(target-1);
    }
}
